src/storage/txn/store.rs

```diff
@@ -58,6 +59,79 @@ pub trait Scanner: Send {
     fn take_statistics(&mut self) -> Statistics;
 }
 
+pub trait TxnEntryStore: Send {
+    /// The scanner type returned by `scanner()`.
+    type Scanner: TxnEntryScanner;
+
+    /// Retrieve a scanner over the bounds.
+    fn entry_scanner(
+        &self,
+        _lower_bound: Option<Key>,
```

Comment:
How about rename to `lower_bound` and `upper_bound`? I think it is more suitable to our code style.